src/compat/attribute.h

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2022 The Bitcoin Core developers
```

Comment:
We already have src/attributes.h. Any reason this is another file?

Reply:
Thanks!. [Updated](https://github.com/bitcoin/bitcoin/pull/24773#issuecomment-1329464328).